prov/efa/test/efa_unit_test_cq.c

```diff
@@ -301,11 +302,20 @@ static void test_impl_rdm_cq_read_unknow_peer_ah(bool remove_peer)
 	struct efa_unit_test_buff recv_buff;
 	int ret;
 
+	if (support_efadv_cq) {
+		g_efa_unit_test_mocks.efadv_create_cq = &efa_mock_efadv_create_cq_with_ibv_create_cq_ex;
```

Comment:
If we want to test the case that support efadv cq, should not we use the real `efadv_create_cq` here?

Reply:
We don't have to, because we also mock `efadv_cq->wc_read_sgid` on L356, so that we can use `ibv_create_cq_ex` instead of the real `efadv_create_cq`.

Using the real `efadv_create_cq` is also problematic in this case (ideally we want to detect the kernel module and set `support_efadv_cq`, but this is not easy). If we use the real `efadv_create_cq` we can get different test results depending on where we run the test:
* If the kernel supports EFA DV CQ and we set `support_efadv_cq = true`, then the test will pass
* If the kernel does **NOT** supports EFA DV CQ and we set `support_efadv_cq = true`, then the test will fail
So we want to avoid this uncertainty by mocking out `efadv_create_cq` entirely.

Reply:
Make sense. Can you add these reasonings as comments.

Reply:
Sure. Will address.

Reply:
Addressed.

Reply:
I did not see the comments?

Reply:
Comment was in the wrong place. Updated again.